addons/sale_stock/static/src/js/qty_at_date_widget.js

```diff
@@ -0,0 +1,74 @@
+var QWeb = core.qweb;
+
+var Widget = require('web.Widget');
+var Context = require('web.Context');
+var data_manager = require('web.data_manager');
+var widget_registry = require('web.widget_registry');
+
+var QtyAtDateWidget = Widget.extend({
+    template: 'sale_stock.qty_at_date',
+    events: _.extend({}, Widget.prototype.events, {
+        'click .fa-info-circle': '_onClickButton',
+    }),
+
+    /**
+     * @override
+     * @param {Widget|null} parent
+     * @param {Object} params
+     */
+    init: function (parent, params) {
+        this.so_line_id = params.data.id;
+        this.data = params.data;
+        this.qty_at_date = params.data.qty_at_date; // Needed in template
+        this.qty_to_delivered = params.data.product_uom_qty - params.data.qty_delivered;
+        this._super(parent);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+    _onClickButton: function (ev) {
+        ev.stopPropagation();
+        var self = this;
+        if (this.so_line_id) {
+            var $modal = $(QWeb.render('sale_stock.QtyDetailModal', {
+                widget: this,
+                data: {
+                    product_name: this.data.product_id.data.display_name,
+                    qty_at_date: this.qty_at_date,
+                    qty_on_hand: this.data.qty_on_hand,
+                    qty_available: this.data.qty_available,
+                    delivery_date: this.data.scheduled_date.add(this.getSession().getTZOffset(this.data.scheduled_date), 'minutes'),
+                    uom_name: this.data.product_uom.data.display_name,
+                    warehouse_view_id: this.data.warehouse_view_id
+                }
+            }));
+            $modal.on('click', '.action_forecasted_quantity', function () {
+                $modal.modal('hide');
+                data_manager.load_action('stock.report_stock_quantity_action').then(function (action) {
+                    // Change action context to choose a specific date and product
+                    // As grid_anchor is set to now() by default in the data, we need
+                    // to load the action first, change the context then launch it via do_action
+                    // additional_context cannot replace a context value, only add new
+                    var additional_context = {}
+                    additional_context.grid_anchor = self.data.scheduled_date.format('YYYY-MM-DD');
+                    additional_context.search_default_product_id = self.data.product_id.data.id;
+                    action.context = new Context(action.context, additional_context);
+                    action.domain = [['location_id', 'child_of', self.data.warehouse_view_id.res_id]];
+                    self.do_action(action);
+                });
+            });
+            $modal.appendTo($('body'));
+            $modal.modal('toggle');
```

Comment:
can't you use web.Modal (follower.js line 369)
you won't need to work "low level" with the modal